stacker/util.py

```diff
@@ -366,3 +369,88 @@ def read_value_from_path(value):
         with open(relative_path) as read_file:
             value = read_file.read()
     return value
+
+
+def sanitize_git_path(uri, ref='HEAD'):
+    """Takes a git URI and ref and converts it to a directory safe path
+
+    Args:
+        uri (string): git URI (e.g. git@github.com:remind101/stacker_blueprints.git)
+        ref (string): optional git ref to be appended to the path
+
+    """
+    if uri.endswith('.git'):
+        dir_name = uri[:-3]  # drop .git
+    else:
+        dir_name = uri
+    dir_name = dir_name.replace('@', '_').replace('/', '_').replace(':', '_')
+    dir_name += "-%s" % ref
+    return dir_name
+
+
+def fetch_git_package(config, package_cache):
+    """Makes a remote git repository available for local use
+
+    Args:
+        config (dict): Dictionary of git repo configuration
+        package_cache (string): Directory in which to store cached packages
+
+    """
+    # Sanitize directory name
+    if 'ref' in config:
+        dir_name = sanitize_git_path(uri=config['uri'], ref=config['ref'])
+    else:
+        dir_name = sanitize_git_path(uri=config['uri'])
+    cached_dir_path = os.path.join(package_cache, dir_name)
+
+    # Clone the repo if it doesn't already exist; otherwise refresh it
+    if not os.path.isdir(cached_dir_path):
+        tmp_dir = tempfile.mkdtemp(prefix='stacker')
+        tmp_repo_path = os.path.join(tmp_dir, dir_name)
+        repo = Repo.clone_from(config['uri'], tmp_repo_path)
```

Comment:
May want to throw all of this in a `try/finally` block to clean up the `tmp_repo_path`